Makefile

```diff
@@ -83,8 +83,10 @@ cross:
 docker: cross
 	cd build && docker build -t $(IMAGE_NAME):$(GIT_HASH) .
 
-release: check test docker
+release-hash: check test docker
 	docker push $(IMAGE_NAME):$(GIT_HASH)
+
+release: check test docker
```

Comment:
Tags are cheap and are the only immutable entities in docker so I don't think we should remove it from the release (OK to have it as an extra for testing). Moreover `release` below depends on the image with the tag to be created... `docker tag $(IMAGE_NAME):$(GIT_HASH) $(IMAGE_NAME):latest`.